common.py

```diff
@@ -57,25 +57,32 @@ def __init__(self, credentials, ca_cert):
         self.private_key = creds['private_key']
         self.login_endpoint = creds['login_endpoint']
         self.verify = False
+        self.auth_header = None
+        self.expiry = 0
         if ca_cert:
             self.verify = ca_cert
 
     def __call__(self, auth_request):
-        payload = {
-            'uid': self.uid,
-            'exp': int(time.time()) + 3600,
-        }
-        token = jwt.encode(payload, self.private_key, 'RS256')
-
-        data = {
-            'uid': self.uid,
-            'token': token.decode('ascii'),
-        }
-        r = requests.post(self.login_endpoint, json=data, verify=self.verify)
-        r.raise_for_status()
-
-        auth_header = 'token=' + r.cookies['dcos-acs-auth-cookie']
-        auth_request.headers['Authorization'] = auth_header
+        if not self.auth_header or int(time.time()) >= self.expiry:
```

Comment:
About the condition `int(time.time()) >= self.expiry`: maybe add some tolerance interval here, so that we fetch the new authentication token maybe 10 minutes earlier than necessary, to prevent a type of face condition that could result in an unnecessary 401 response.